vim_menutrans_helper.py

```diff
@@ -0,0 +1,196 @@
+    origin = origin.replace('\\"', '"')
+    origin = origin.replace("\\'", "'")
+    origin = origin.replace('\\t', '\t')
+    origin = origin.replace('\\\\', '\\')
+    return origin
+
+# unescape single quotes
+def unescape_single_quotes(origin):
+    origin = origin.replace("''", "'")
+    return origin
+
+# get the diffence between untranslated file and translated file
+def makeUnTranslatedDic(untranslated_file, untranslated_dict):
+    # deal with the file need to be translated
+    with open(untranslated_file, encoding="latin1") as f1:
+        for line_number, line in enumerate(f1):
+            tobe_translated_words = []
+
+            line = line.strip()
+            line = line.rstrip()
+
+            temp_word_list = []
+            pre_end = -1
+            top_string = line.split()
+            if len(top_string) > 0 and (top_string[0] == "exe" or top_string[0] == "exec" or top_string[0] == "execute"):
+                # extract content inside quotes from line
+                for ite in re.finditer(r"'(?:''|[^'])*'|(?:\"(?:(?:\\.)|[^\"])*\")", line):
+                    mid_part = line[pre_end + 1: ite.start() - 1].strip()
+                    if not ((len(mid_part) == 1 and mid_part[0] == ".") or len(mid_part) == 0 or pre_end == -1):
+                        temp_word_list.append("5")
+                    pre_end = ite.end()
+                    temp_word_list.append(line[ite.start() + 1: ite.end() - 1])
+
+                temp_line = ""
+                for word in temp_word_list:
+
+                    if len(word) == 0:
+                        continue
+                    if word[0] == "'":
+                        word = unescape_single_quotes(word)
+                    else:
+                        word = unescape_double_quotes(word)
+                    temp_line += word
+                line = temp_line
+
+            word_list = []
+            # split line string according to blank without backslash or tab ahead
+            for ite in re.finditer(r"(?:(?:\\.)|[^ \t])+", line):
+                word_list.append(line[ite.start(): ite.end()])
+
+            if len(word_list) == 0 or word_list[0] == '"':
+                continue
+            index = 0
+            # menu_item with "disable" or enable followed close behind should be ignored
+            if word_list[0] in menu_item and len(word_list) > 1 and (word_list[1] == "disable" or word_list[1] == "enable"):
+                continue
+
+            if word_list[0] == 'sil!':
+                index += 1
+            # blocks needs to be translated without being splited by '.'
+            if index < len(word_list) and word_list[index] in menu_item:
+                tobe_translated_blocks = ""
+                index += 1
+                while index < len(word_list):
+                    if word_list[index] == '<silent>' or word_list[index] == '<script>' or (word_list[index][0].isdigit() and not word_list[index - 1][0].isdigit()):
```

Comment:
Line too long. Please wrap it at some proper places to make each physical line shorter than 80 characters. There are a few other long lines within this file. Please wrap them all.